typeinfo.go

```diff
@@ -237,7 +237,7 @@ func decodeStructValue(d *Decoder, v reflect.Value) error {
 //------------------------------------------------------------------------------
 
 func encodeCustomValue(e *Encoder, v reflect.Value) error {
-	if v.IsNil() {
+	if v.Kind() == reflect.Ptr && v.IsNil() {
```

Comment:
"IsNil reports whether its argument v is nil. The argument must be a chan, func, interface, map, pointer, or slice value; if it is not, IsNil panics."

Will do a proper fix tomorrow.